src/DotNetWorker.Grpc/FunctionMetadata/FunctionMetadataRpcExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Text.Json;
+using Google.Protobuf.Collections;
+using Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;
+using Microsoft.Azure.Functions.Worker.Grpc.Messages;
+
+namespace Microsoft.Azure.Functions.Worker.Grpc.FunctionMetadata
+{
+    internal static class FunctionMetadataRpcExtensions
+    {
+        internal static MapField<string, BindingInfo> GetBindingInfoList(this IFunctionMetadata funcMetadata)
+        {
+            if (funcMetadata is RpcFunctionMetadata rpcFuncMetadata)
+            {
+                return rpcFuncMetadata.Bindings;
+            }
+
+            MapField<string, BindingInfo> bindings = new MapField<string, BindingInfo>();
+            var rawBindings = funcMetadata.RawBindings;
+
+            if(rawBindings.Count == 0)
+            {
+                throw new FormatException("At least one binding must be declared in a Function.");
+            }
+
+            foreach (var bindingJson in rawBindings)
+            {
+                var binding = JsonSerializer.Deserialize<JsonElement>(bindingJson);
```

Comment:
FYI: I opened #1057 to make this code trim friendly. We can address in a future PR as needed.

We need to replace the use of JsonElement with a type.

    public class HttpBindingInfo
    {
        // add the needed props
    }

    [JsonSourceGenerationOptions(WriteIndented = true)]
    [JsonSerializable(typeof(HttpBindingInfo))]
    public partial class SourceGenerationContext : JsonSerializerContext
    {
    }
and

    var binding = JsonSerializer.Deserialize<HttpBindingInfo>(bindingJson,
                                       SourceGenerationContext.Default.HttpBindingInfo);